client/my-sites/checkout/src/components/cost-overrides-list.tsx

```diff
@@ -323,17 +324,20 @@ export function CouponCostOverride( {
 	const { formStatus } = useFormStatus();
 	const isDisabled = formStatus !== FormStatus.READY;
 	const isOnboardingAffiliateFlow = useSelector( getIsOnboardingAffiliateFlow );
+	const [ , experimentAssignment ] = useExperiment( 'calypso_checkout_hide_coupon_box' );
 
 	if ( ! responseCart.coupon || ! responseCart.coupon_savings_total_integer ) {
 		return null;
 	}
 
-	// translators: The label of the coupon line item in checkout, including the coupon code
+	// translators: The label of the coupon line item in checkout, including the coupon coden
+	const couponLabel = translate( 'Coupon: %(couponCode)s', {
+		args: { couponCode: responseCart.coupon },
+	} );
+
 	const label = isOnboardingAffiliateFlow
 		? getAffiliateCouponLabel()
-		: translate( 'Coupon: %(couponCode)s', {
-				args: { couponCode: responseCart.coupon },
-		  } );
+		: getCouponLabel( couponLabel as string, experimentAssignment?.variationName || null );
```

Comment:
Should the new label be shown if the product is a part of the ignored slugs?

Reply:
Yes it should, the ignored slugs only affect if the coupon box is hidden